### PR TITLE
Implement most of frustum

### PR DESCRIPTION
**Pull request by jtenner**
Left out `Sprite` because introduced meshes and materials which caused a big problem (overflowing my stdout with diagnostics)

Left out intersectsObject because it requires meshes and geometries. (We should probably revert the geometry commits now right?)

### CODE REVIEW

**Comment by trusktr** [comment]
>  (We should probably revert the geometry commits now right?)

Or just fix up what is needed. Basically any `material` and `geometry` properties are for the `Mesh` class

**Comment by trusktr** [comment]
Good to merge now if you want as tests are passing. No biggie if it isn't complete yet. The whole thing is WIP!

**Comment by jtenner** [comment]
I'm okay with getting around to this tomorrow.

File: src/as/math/Matrix4.ts [hunk 1]
```diff
@@ -918,35 +918,35 @@ export class Matrix4 /*implements Matrix*/ {
...
+		te[9] = <f32>0
+		te[13] = -y
+		te[2] = <f32>0
+		te[6] = <f32>0
+		te[10] = <f32>-2 * p
+		te[14] = -z
+		te[3] = <f32>0
+		te[7] = <f32>0
+		te[11] = <f32>0
+		te[15] = <f32>1
```
Comment (trusktr) [inline src/as/math/Matrix4.ts:946]: Is it possible to avoid the `<f32>` casts somehow?

Comment (jtenner) [reply src/as/math/Matrix4.ts:946]: I wouldn't have added them unless they were needed :)

**Pull request merged**